[WIP] Av1 support fixes

kodabb opened the pull request:
Hey @VFR-maniac I tried restoring your branch to provide AV1 support to L-SMASH, but I'm stuck on a couple of points, do you think you could chime in?

- in `av1_get_access_unit()` how access the frame?
- what are the macro values for `MAX_TIME_{ROWS,COLS,WIDTH,AREA}`?
- what is how to get to the summary needed by `av1_parse_sequence_header()` in `lsmash_setup_av1_specific_parameters_from_access_unit()`?
- what is the implementation of `av1_motion_field_estimation()`, `tile_log2`, and `av1_is_shown_frame`?

thanks for any insight

dwbuiten commented:
> Hey @VFR-maniac I tried restoring your branch to provide AV1 support to L-SMASH, but I'm stuck on a couple of points, do you think you could chime in?

Looking forward to AV1 in L-SMASH~

>* in `av1_get_access_unit()` how to access the frame?

What do you mean by 'access the frame'? Why do you need to?

Do you mean setting sample properties (which should be straightforward)?

>* what are the macro values for `MAX_TIME_{ROWS,COLS,WIDTH,HEIGHT,AREA}`?

I assume you mean `MAX_TILE_*` and not `MAX_TIME_*`.

They are all here: https://aomediacodec.github.io/av1-spec/#symbols-and-abbreviated-terms

>* how to get to the summary needed by `av1_parse_sequence_header()` in `lsmash_setup_av1_specific_parameters_from_access_unit()`?
>* what is the implementation of `av1_motion_field_estimation()`, `tile_log2`, and `av1_is_shown_frame`?

* `av1_motion_field_estimation`: Section 5.9.2 shows this function being called in the uncompressed header parser: https://aomediacodec.github.io/av1-spec/#motion-field-estimation-process
* `tile_log2`: https://aomediacodec.github.io/av1-spec/#tile-size-calculation-function
* `av1_is_shown_frame`: I'm not sure what the intended use-case of this function is in the end, but presumably it is based on the various related header fields. Only @VFR-maniac could answer, I think.

kodabb commented:
> What do you mean by 'access the frame'? Why do you need to?
>
> Do you mean setting sample properties (which should be straightforward)?

No, I mean that in `av1_get_access_unit()` there are two functions that use `frame` internally, `av1_parse_frame_header()` and `av1_parse_frame()`. After that there are two variables `MaxRenderWidth` and `MaxRenderHeight` that are derived using frame properties. Finally the function signatue itself cannot change because its pointer is used in the importer.

> I assume you mean MAX_TILE_* and not MAX_TIME_*.

yes ^^

> They are all here: https://aomediacodec.github.io/av1-spec/#symbols-and-abbreviated-terms

thanks, added to the PR

>    av1_motion_field_estimation: Section 5.9.2 shows this function being called in the uncompressed header parser: https://aomediacodec.github.io/av1-spec/#motion-field-estimation-process

`// TODO`

>    tile_log2: https://aomediacodec.github.io/av1-spec/#tile-size-calculation-function

added

>    av1_is_shown_frame: I'm not sure what the intended use-case of this function is in the end, but presumably it is based on the various related header fields. Only @VFR-maniac could answer, I think.

also av1_parse_sequence_header() without parameters is puzzling, I hope @VFR-maniac can comment
